modules/core/04-channel/keeper/timeout.go

```diff
@@ -150,13 +150,6 @@ func (k Keeper) TimeoutExecuted(
 
 	k.deletePacketCommitment(ctx, packet.GetSourcePort(), packet.GetSourceChannel(), packet.GetSequence())
 
-	if channel.Ordering == types.ORDERED {
-		channel.State = types.CLOSED
-		k.SetChannel(ctx, packet.GetSourcePort(), packet.GetSourceChannel(), channel)
-		emitChannelClosedEvent(ctx, packet, channel)
-	}
-
-	// TODO: handle situation outlined in https://github.com/cosmos/ibc-go/issues/4454
 	// if an upgrade is in progress, handling packet flushing and update channel state appropriately
 	if channel.State == types.STATE_FLUSHING {
 		counterpartyUpgrade, found := k.GetCounterpartyUpgrade(ctx, packet.GetSourcePort(), packet.GetSourceChannel())
```

Comment:
Let's say the timeout on `counterpartyUpgrade` is not set yet (TRY has been called, we are in state `FLUSHING` but counterparty timeout is not yet set), but we are in the process of timing out a packet on an ordered channel. 

We would end up skipping the code in `timeout.IsValid()`, timing out the packet and closing the channel, but the upgrade state would still remain in store.

Reply:
guess we should be able to conditionally delete in the `if channel.Ordering == types.ORDERED` block yea?

Reply:
really nice catch, I guess then the situation we can be in is that

timeout is invalid, LastSequenceSend is 0, but upgradeFields are set. We could use this as the check for also triggering the abort.